third_party/airflow/tests/unit/test_armada_operator.py

```diff
@@ -32,3 +36,22 @@ def test_get_lookout_url(lookout_url_template, job_id, expected_url):
     )
 
     assert operator._get_lookout_url(job_id) == expected_url
+
+
+class TestJobService(unittest.TestCase):
+    def test_cancel_jobs(self, job_id, queue):
+        # Set the job_set_id and queue
+        self.job_set_id = job_id
+        self.queue = queue
+
+        # Call the on_kill function
+        self.on_kill()
+
+        # Cancel the jobs
+        response = self.armada_client.cancel_jobset(queue=self.queue, job_set_id=self.job_set_id)
```

Comment:
I don’t think this will work like you think.  Armada is not running in the unit test so cancel shouldn’t work.  I would go with mocking the cancel_jobs function

Reply:
ok, thank you @kannon92

Reply:
@kannon92  pls review the changes!!

Reply:
Can you check the test failures?  And why is this separate from your other PR for on_kill()?

Reply:
I thought it would be easy to review changes if I create new PR, ok I will close this one and merge changes with the previous pr!!